### PR TITLE
dev-cpp/asio

### PR DESCRIPTION
**Pull request by DarthGandalf**

### CODE REVIEW

File: dev-cpp/asio/asio-1.30.1.ebuild [hunk 1]
```diff
@@ -23,6 +23,12 @@ DEPEND="
 "
 BDEPEND="virtual/pkgconfig"
 
+PATCHES=(
+	"${FILESDIR}/asio-1.30.1-pkgconfig.patch"
+)
+
+S="${WORKDIR}/asio-asio-${PV//./-}/asio"
```
Comment (SoapGentoo) [inline dev-cpp/asio/asio-1.30.1.ebuild:30]: put this after `SRC_URI`

File: dev-cpp/asio/files/asio-1.30.1-pkgconfig.patch [hunk 2]
```diff
@@ -0,0 +1,13 @@
+Install .pc file to /usr/share instead of /usr/lib because there are no .so files
+
+--- a/Makefile.am
++++ b/Makefile.am
+@@ -1,7 +1,7 @@
+ AUTOMAKE_OPTIONS = foreign dist-bzip2 dist-zip
+ 
+ pkgconfig_DATA = asio.pc
+-pkgconfigdir = $(libdir)/pkgconfig
++pkgconfigdir = $(datadir)/pkgconfig
```
Comment (SoapGentoo) [inline dev-cpp/asio/files/asio-1.30.1-pkgconfig.patch:10]: delete the entire line, and call `PKG_NOARCH_INSTALLDIR` in `configure.ac`, and then use `noarch_pkgconfig_DATA` above

**Review by SoapGentoo — CHANGES_REQUESTED** [review]

File: dev-cpp/asio/files/asio-1.30.1-pkgconfig.patch (hunk 2 above)
Comment (DarthGandalf) [reply dev-cpp/asio/files/asio-1.30.1-pkgconfig.patch]: Cool

File: dev-cpp/asio/asio-1.30.1.ebuild (hunk 1 above)
Comment (DarthGandalf) [reply dev-cpp/asio/asio-1.30.1.ebuild]: Done

File: dev-cpp/asio/files/asio-1.30.1-pkgconfig.patch [hunk 3]
```diff
@@ -0,0 +1,24 @@
+Install .pc file to /usr/share instead of /usr/lib because there are no .so files
+
+--- a/Makefile.am
++++ b/Makefile.am
+@@ -1,7 +1,6 @@
+ AUTOMAKE_OPTIONS = foreign dist-bzip2 dist-zip
+ 
+-pkgconfig_DATA = asio.pc
+-pkgconfigdir = $(libdir)/pkgconfig
++noarch_pkgconfig_DATA = asio.pc
```
Comment (juippis) [inline dev-cpp/asio/files/asio-1.30.1-pkgconfig.patch:10]: Heh, this scared me:
```
 FILES:-usr/lib64/pkgconfig/asio.pc
 FILES:+usr/share/pkgconfig/asio.pc
```
but I guess it's by the plan.

**Review by juippis — APPROVED** [review]

**Pull request closed without merging**